src/Pages/LoginPage.ts

```diff
@@ -0,0 +1,125 @@
+@injectable()
+export class LoginPage extends BasePage implements PageInterface {
+
+    @inject(ContainerName.Container)
+    public readonly container!: Container;
+
+    /**
+     * Selector to LoginPage
+     *
+     * @type {LoginSelectorType}
+     */
+    public $s: LoginSelectorType = loginSelector;
+
+    public hasLogged: boolean = false;
+
+    /**
+     * Execute this step
+     *
+     * @returns {Promise<void>}
+     */
+    public async execute(): Promise<void> {
+        await this.start(async () => {
+            await this.preStart();
+            await this.homeStep();
+            await this.loginStep();
+        });
+    }
+
+    /**
+     * Process home
+     *
+     * @returns {Promise<void>}
+     */
+    public async homeStep(): Promise<void> {
+        // eslint-disable-next-line unicorn/numeric-separators-style
+        await this.page.goto("https://www.latamairlines.com/br/pt", { timeout: 15000 });
+        try {
+            await expect(this.page.locator(this.$s.cookiesBtn)).toBeVisible({ timeout: 5000 });
+            await this.clickBtn(this.$s.cookiesBtn);
+        } catch { /* Ignore Error */ }
+
+        await this.clickBtn(this.$s.loginBtn);
+    }
+
+    /**
+     * Process login
+     *
+     * @returns {Promise<void>}
+     */
+    // eslint-disable-next-line max-statements
+    public async loginStep(): Promise<void> {
+        const successHttp = 200;
+        let login = await this.config.get(ConfigName.LATAM_LOGIN);
+        login ??= "";
+
+        let password = await this.config.get(ConfigName.LATAM_PASSWORD);
+        password ??= "";
+
+        await this.fillInput(this.$s.emailInput, login);
+        let responsePromise = this.page.waitForResponse(`https://www.latamairlines.com/bff/auth/v1/user/auth/${login}/search`);
+        await this.clickBtn(this.$s.primaryBtn);
+        let response = await responsePromise;
+        if (response.status() === successHttp) {
+            try {
+                // eslint-disable-next-line unicorn/numeric-separators-style
+                await expect(this.page.locator(this.$s.passwordInput)).toBeVisible({ timeout: 15000 });
+                await this.fillInput(this.$s.passwordInput, password);
+                responsePromise = this.page.waitForResponse("https://accounts.latamairlines.com/usernamepassword/login");
+                await this.page.locator(this.$$s.loginSelector.primaryBtn).first().click();
+                response = await responsePromise;
+                this.container.bind(ContainerName.LoginResponse).toConstantValue(response);
+            } catch { /* Ignore Error */ }
+        } else {
+            this.container.bind(ContainerName.LoginResponse).toConstantValue(response);
+        }
+    }
+
+    /**
+     * Click On Btn
+     *
+     * @param {string} selector btn selector
+     * @returns {Promise<void>}
+     */
+    public async clickBtn(selector: string): Promise<void> {
+        return this.page.locator(selector).first().click();
```

Comment:
seu seletor deve ser unico então nunca vai precisar do first, exceto em seletetores de lista, como passageiros.
```suggestion
        return this.page.click(selector); // isso torna essa função e as demais desnecessarias
```